### ISSUE TITLE
NativeBase's Text component does't support nested styling

### ISSUE THREAD

**Issue by trajano** [GeekyAnts/NativeBase#4152]
The original one https://github.com/GeekyAnts/NativeBase/issues/610 was closed.  However, the most recent post to it https://github.com/GeekyAnts/NativeBase/issues/610#issuecomment-921715804 confirms I am not alone facing the issue.

In my case I tested the following with a custom font family...

        <Text bold italic underline>
          bold Italic underline
        </Text>{" "}
        <Text bold>
          <Text italic>
            <Text underline>
              <Text>bold Italic underline NESTED</Text>
            </Text>
          </Text>
        </Text>{" "}

Here is the output

![image](https://user-images.githubusercontent.com/110627/135695837-ad8fcf63-2b15-4af6-98c4-1974ea4fbab4.png)

When looking at the children, it appears to be putting the regular font only.

**Comment by trajano** [comment]
I was able to create [a snack](https://snack.expo.dev/@trajano/not-nesting-correctly) that demonstrates this

```
    <>
      <View>
        <Text fontSize="72">Taj</Text>
      </View>
      <View>
        <Text fontSize="32">
          <Text bold>Taj Mahal</Text> is in Agra.
        </Text>
      </View>
      <View>
        <Text bold underline italic>
          Taj Mahal
        </Text>
      </View>
      <View>
        <Text bold>
          <Text underline>
            <Text italic>
              <Text>Taj Mahal</Text>
            </Text>
          </Text>
        </Text>
      </View>
      <View>
        <Text underline>
          <Text bold>
            <Text italic>
              <Text>Taj Mahal</Text>
            </Text>
          </Text>
        </Text>
      </View>
 ```

![image](https://user-images.githubusercontent.com/110627/135697188-f9f2c51e-2813-4cce-817c-c9f3a4e9888f.png)

Considering all this and I want to work with Markdown like content which will have a lot of nested stylings, I think the NativeBase `Text` needs work.

**Comment by trajano** [comment]
I'm suspecting it is because of https://github.com/GeekyAnts/NativeBase/blob/master/src/components/primitives/Text/index.tsx#L11 and it may be forcing the styles but not doing the inhertance.

**Comment by trajano** [comment]
I tried to implement using `Context` and `Text` from React-Native https://github.com/trajano/text-custom

**Comment by intergalacticspacehighway** [comment]
@trajano thanks for reporting! A fix is pushed in [3.2.2-rc.1](https://github.com/GeekyAnts/NativeBase/releases/tag/v3.2.2-rc.1)
